src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerService.java

```diff
@@ -113,6 +117,12 @@ public void updateGeneralPartner(String generalPartnerId, GeneralPartnerDataDto
         repository.save(generalPartnerDaoAfterPatch);
     }
 
+    public GeneralPartnerDto getGeneralPartner(Transaction transaction, String submissionId) throws ServiceException {
+        checkGeneralPartnerIsLinkedToPartnership(transaction, submissionId);
+        var generalPartnerDao = repository.findById(submissionId).orElseThrow(() -> new ResourceNotFoundException(String.format("General partner submission with id %s not found", submissionId)));
+        return mapper.daoToDto(generalPartnerDao);
```

Comment:
I think you have the wrong id to search with, it should be the generalPartnerId rather than the submissionId

Reply:
The submission id is the general partner id, I was going to call it that originally but decided to go with the pattern that was already there is that had already been reviewed and merged.

Reply:
Though looking at update it would seem we have two different approaches emerging here.

Reply:
I'll call it submission in the controller but generalPartnerId in the service.

Reply:
don't do that as the submissionID comes after /submission/ in the url, but the url for general partner does not contain /submission/

Reply:
Yes that's what I meant

Reply:
Let's be consistent within this PR I think. Do we need a call @mattch1 ?

Reply:
Updated parameter name.